includes/class-wc-payments-webhook-processing-service.php

```diff
@@ -202,6 +202,17 @@ private function process_webhook_refund_updated( $event_body ) {
 			),
 			$refund_id
 		);
+
+		// Delete refund based on WCPay refund id.
+		$wc_refunds = $order->get_refunds();
+		foreach ( $wc_refunds as $wc_refund ) {
+			$wcpay_refund_id = $wc_refund->get_meta( '_wcpay_refund_id', true );
+			if ( $refund_id === $wcpay_refund_id ) {
+				$wc_refund->delete();
+				break;
```

Comment:
@achyuthajoy - I am working on this issue https://github.com/automattic/woocommerce-payments/issues/3814#issuecomment-1066691553 and since you're working here, I think it would integrate it for handling `charge.refund.updated` event. 

More specifically, if there is no refund ID having the `refund_id` from the webhook data, let's stop right away without processing further.

Reply:
Sure @dat.

Just wanted to confirm that we need to skip processing the refund event, if `_wcpay_refund_status` is already marked as failed or if we can't find a refund associated with the wcpay refund id. Correct?

Reply:
Thank you. 

> if we can't find a refund associated with the wcpay refund id.

It would be only this condition :)

Reply:
After discussing further, we decided to withdraw this idea and use the same approach `order_note_exists` in this PR https://github.com/Automattic/woocommerce-payments/pull/3975

Going to mark this resolved.